From e83e7820ed8a10d3f15d6627e542674283cc577a Mon Sep 17 00:00:00 2001
From: Sathyanarayanan Kuppuswamy <sathyanarayanan.kuppuswamy@intel.com>
Date: Wed, 30 Oct 2013 21:13:54 -0700
Subject: [PATCH] device: Add power_hal suspend device attribute


Added following macro to help drivers export their
suspend/resume functionality via sysfs attributes.

DEVICE_POWER_HAL_SUSPEND_ATTR(_store) - Creates a WO
device attribute named 'dev_attr_power_hal_suspend' with
given set function.

Signed-off-by: Sathyanarayanan Kuppuswamy <sathyanarayanan.kuppuswamy@intel.com>
---
 include/linux/device.h |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/linux/device.h b/include/linux/device.h
index c0a126125325..45e56a01ef58 100644
--- a/include/linux/device.h
+++ b/include/linux/device.h
@@ -516,6 +516,9 @@ ssize_t device_store_bool(struct device *dev, struct device_attribute *attr,
 #define DEVICE_ATTR_IGNORE_LOCKDEP(_name, _mode, _show, _store) \
 	struct device_attribute dev_attr_##_name =		\
 		__ATTR_IGNORE_LOCKDEP(_name, _mode, _show, _store)
+#define DEVICE_POWER_HAL_SUSPEND_ATTR(_store) \
+	DEVICE_ATTR(power_HAL_suspend, S_IWUSR|S_IWGRP, \
+		NULL, _store)
 
 extern int device_create_file(struct device *device,
 			      const struct device_attribute *entry);
-- 
1.7.9.5

